### PR TITLE
Create Shipment and Transfers Tables

### PR DESCRIPTION
**Pull request by vahidbazzaz**
This PR contains the migrations required for creation of shipment and transfer tables. 

Related PR: https://github.com/boxwise/boxtribute/pull/285

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[BUG/ISSUE: Shipment and transfers tables do not currently exist in the live staging / production databases](https://trello.com/c/YB02E4tM)

File: db/migrations/20220701094045_add_transfer_agreement_table.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                'after' => 'transfer_agreement_id',
+            ])
+            ->addForeignKey('source_organisation_id', 'organisations', 'id', [
+                'update' => 'CASCADE',
+            ])
+            ->addForeignKey('target_organisation_id', 'organisations', 'id', [
+                'update' => 'CASCADE',
+            ])
+            ->addForeignKey('requested_by', 'cms_users', 'id', [
+                'update' => 'CASCADE',
```
Comment (pylipp) [inline db/migrations/20220701094045_add_transfer_agreement_table.php:87]: Does it make sense to not have the `delete` option set? I know you followed the peewee model definition so I wonder what I was thinking to not define `delete` there :thinking:

File: db/migrations/20220701094148_add_shipment_table.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                'null' => false,
+                'signed' => false,
+                'after' => 'source_base_id',
+            ])
+            ->addColumn('transfer_agreement_id', 'integer', [
+                'null' => false,
+                'signed' => false,
+                'after' => 'target_base_id',
+            ])
+            ->addColumn('status', 'string', [
```
Comment (pylipp) [inline db/migrations/20220701094148_add_shipment_table.php:34]: See above. Requesting to change to `state` here and in line 42 :)

File: db/migrations/20220701094045_add_transfer_agreement_table.php (hunk 1 above)
Comment (pylipp) [reply db/migrations/20220701094045_add_transfer_agreement_table.php:87]: Ahh never mind, the `requested_by` field is defined as `NOT NULL`, so `ON DELETE SET NULL` does not make sense.

File: db/migrations/20220701094045_add_transfer_agreement_table.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            'null' => false,
+            'signed' => false,
+            'after' => 'id',
+        ])
+            ->addColumn('target_organisation_id', 'integer', [
+                'null' => false,
+                'signed' => false,
+                'after' => 'source_base_id',
+            ])
+            ->addColumn('status', 'string', [
```
[the remark this answers is not in the document]
Comment (vahidbazzaz) [reply db/migrations/20220701094045_add_transfer_agreement_table.php]: @pylipp if we want to enforce ENUM values, then we need to use it as DB constraint and we can use Check constraint but is not supported in this version Phinx and then we need to run custom query for it. So I think it make sense to leave it as string with default value.

File: db/migrations/20220701094045_add_transfer_agreement_table.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            ->addColumn('valid_from', 'datetime', [
+                'null' => false,
+                'after' => 'terminated_by',
+            ])
+            ->addColumn('valid_until', 'datetime', [
+                'null' => true,
+                'after' => 'valid_from',
+            ])
+            ->addColumn('comment', 'text', [
+                'null' => false,
```
Comment (pylipp) [inline db/migrations/20220701094045_add_transfer_agreement_table.php:77]: I wonder whether comment should be nullable, just as in the `stock` table. What do you think?

Comment (vahidbazzaz) [reply db/migrations/20220701094045_add_transfer_agreement_table.php:77]: @pylipp thanks, thats totally make sense, I will make it nullable, please update this in your end too  ;)

**Review by pylipp — APPROVED** [review]

**Pull request merged**